### DISCUSSION TITLE
DSC PC1500 ESP32 Homebridge Homekit fork created

### DISCUSSION THREAD

**Discussion by polishedmarvin** [taligentx/dscKeybusInterface#171] [dscKeybusInterface Discussion]
I have created a fork for my implementation of a DSC PC1500 Keybus Reader for ESP32 Homebridge Homekit 

@polishedmarvin/dscKeybusInterface/examples/ESP32/PC1500KeybusReader

   Adapted from dougkpowers/pc1550-interface
   Adapted from taligentx/dscKeybusInterface for Homebridge

**Comment by taligentx** [comment]
Hi @polishedmarvin - Nice!  It's good to see this for the DSC classic series, I'll update the README with a link to your implementation.

It also should be possible to get data from the PC1500 by using interrupts for the clock and data lines, similar to this library - the advantage is that the sketch would not need to block and halt all other processing while collecting data from the panel (as is currently done in dougkpowers/pc1550-interface).

It's been on the TODO list, but if the ESP32 is handling MQTT well with the looping code then that's great news for the PC1500 folks!

**Reply by polishedmarvin** [reply]
Hi @taligentx - thanks for noticing my submission and pointing to it in your README. I hope that someone else will find it helpful to their project. 
I agree that the looping structure is not ideal. Maybe someone will take the next step and develop this into an interrupt based design. As my first coding project since 1988 and no previous C++ experience  I chose to stick close to a design that was proven. 
The ESP32 seems to manage well. I am using the dual core processor version and while I have not verified it, I believe that the second core takes care of high priority work like managing the WiFi connection and MQTT communication.

**Comment by taligentx** [comment]
Hi @polishedmarvin - I've added support for the Classic series (so far, PC1500/PC1550) to the `develop` branch.  This uses the same interrupt-driven data capture as the PowerSeries, and the Homebridge sketch is supported as well.

**Comment by polishedmarvin** [comment]
Great. When I get a chance I will try it out.

**Reply by neohusky** [reply]
Hi @polishedmarvin & @taligentx 
Thank you to both of you for your continued work and interest in this project.Just wondering how easy it would be to port either of your implementations (even the RTOS one) to use esp-homekit-sdk to create as standalone homekit ESP32 device without homebridge?
